base/applications/rapps/loaddlg.cpp

```diff
@@ -982,11 +982,16 @@ CDownloadManager::ThreadFunc(LPVOID param)
         // run it
         if (InfoArray[iAppId].DLType == DLTYPE_APPLICATION)
         {
+            BOOL bZip = (lstrcmpiW(PathFindExtensionW(Path), L".zip") == 0);
```

Comment:
Critics:
1.) Questionable to do this specifically for .zip, because the vast majority of zip fles will open fine with zipfldr. Only those with specific compression algorithms will cause issues
2.) What about .7z / .rar and the like, they will still cause the same issue

**Therefore suggestion1 for workaround:**
If you want to workaround then check for ".exe" / ".com" / ".bat" / ".reg" / ".msi" instead, and if found try to execute them.
If none of such extensions, then open the containing folder.

**Alternative suggestion2 for workaround:**
Try to execute them all, without extension check, if that fails, then do fail **silently** and try to open the containing folder instead.

**Alternative suggestion 3 for a proper fix:**
Find an external compression lib for a large variety of compression formats and use that from within zipfldr.

If neither 1,2,3 is of your taste, then I would suggest to just leave it like it is today.

Reply:
Fixed.